templates/project/_featuresList.html.twig

```diff
@@ -10,43 +10,23 @@
     </thead>
     <tbody>
 
-    {{ form_start(form.projectFeatures) }}
-
-    {% for projectFeature in form.projectFeatures %}
+    {% for featureCategoryIndex, featureCategory in featureCategories %}
         <tr>
-            <th class="w-50" scope="row">
-
-                <div>
-                <a class="collapse-toggler collapsed" data-toggle="collapse" href="#collapseDescription{{ loop.index }}" aria-expanded="false" aria-controls="collapseDescription{{ loop.index }}">
-                    {{ projectFeature.vars.data.feature.name }}
+            <td colspan="4" class="feature-category w-50">
+                <div><a class="collapse-toggler" data-toggle="collapse" href="#collapseDescription{{ featureCategory.id }}" aria-expanded="true" aria-controls="collapseDescription{{ featureCategory.id }}">
+                    Catégorie : {{ featureCategory.name }}
                 </a></div>
-                <div class="collapse" id="collapseDescription{{ loop.index }}">
-                    <div class="mt-3 description small" colspan="3">
-                        {{ form_widget(projectFeature.description,{'attr': {'rows': '3'}}) }}
-                        {{ form_errors(projectFeature.description) }}
-                    </div>
-                </div>
-
-                {{ form_widget(projectFeature.feature, {'attr': {'class': 'hidden-field'}}) }}
-                {{ form_errors(projectFeature.feature) }}
-            </th>
-
-            <td class="text-center">
-                {{ form_widget(projectFeature.category) }}
-                {{ form_errors(projectFeature.category) }}
             </td>
 
-            <td class="text-center input-align">
-                {{ form_widget(projectFeature.day) }} j*h
-                {{ form_errors(projectFeature.day) }}
-            </td>
-
-            <td class="text-center">
-                <button title="Supprimer la fonctionnalité" class="btn btn-sm btn-primary" type="submit" formmethod="post" formaction="{{ path('project_feature_delete', {'id': projectFeature.vars.data.id}) }}">
-                    <i class="fas fa-trash-alt"></i>
-                </button>
-            </td>
         </tr>
+
+        {% for projectFeature in form.projectFeatures %}
+            {% if featureCategory.id == projectFeature.vars.data.category.id %}
+                {{ include('project/_featureLine.html.twig') }}
+            {%  endif %}
+        {% else %}
+            <td>Pas de fonctionnalités enregistrées !</td>
+        {% endfor %}
     {% else %}
         <td>Pas de fonctionnalités enregistrées !</td>
```

Comment:
je ne devrais pas avoir 2 fois le meme message d'erreurs dans 2 for différents :/ c'est sans doute optimisable

Reply:
Corrigé